walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,89 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Sort list by events.
+    ArrayList<Event> eventList = new ArrayList(events);
+    Collections.sort(eventList, (Event e1, Event e2) -> TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
+    
+    Collection<TimeRange> optionalOpenTimes = getOpenTimes(eventList, request, true);
+    Collection<TimeRange> requiredOpenTimes = getOpenTimes(eventList, request, false);
+
+    if (optionalOpenTimes.size() > 0) {
+      return optionalOpenTimes;
```

Comment:
nit: this variable name makes it seem like it's open times for optional attendees only, maybe rename to optionalAndRequiredOpenTimes or optionalIncludedOpenTimes?

Reply:
Ah yes that would sound better thanks!